core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitorActor.scala

```diff
@@ -171,26 +178,57 @@ trait SubmissionMonitor extends FutureSupport with LazyLogging with RawlsInstrum
 
     def abortActiveWorkflows(submissionId: UUID): Future[Seq[(Option[String], Try[ExecutionServiceStatus])]] = {
       datasource.inTransaction { dataAccess =>
-        // look up abortable WorkflowRecs for this submission
-        dataAccess.workflowQuery.findWorkflowsForAbort(submissionId).result
-      }.flatMap { workflowRecs =>
-        Future.traverse(workflowRecs) { workflowRec =>
-          Future.successful(workflowRec.externalId).zip(executionServiceCluster.abort(workflowRec, UserInfo.buildFromTokens(credential)))
+        for {
+          // look up abortable WorkflowRecs for this submission
+          wfRecs <- dataAccess.workflowQuery.findWorkflowsForAbort(submissionId).result
+          submissionRec <- dataAccess.submissionQuery.findById(submissionId).result.map(_.head)
+          submitter <- dataAccess.rawlsUserQuery.load(RawlsUserRef(RawlsUserSubjectId(submissionRec.submitterId))).map(_.get)
+          workspaceRec <- dataAccess.workspaceQuery.findByIdQuery(submissionRec.workspaceId).result.map(_.head)
+        } yield {
+          (wfRecs, submitter, workspaceRec)
         }
+      } flatMap { case (workflowRecs, submitter, workspaceRec) =>
+          for {
+            petSAJson <- samDAO.getPetServiceAccountKeyForUser(workspaceRec.namespace, submitter.userEmail)
+            petUserInfo <- googleServicesDAO.getUserInfoUsingJson(petSAJson)
+            abortResults <- Future.traverse(workflowRecs) { workflowRec =>
+              Future.successful(workflowRec.externalId).zip(executionServiceCluster.abort(workflowRec, petUserInfo))
+            }
+          } yield {
+            abortResults
+          }
+      }
+    }
+
+    def traverseWorkflowOutputs(externalWorkflowIds: Seq[WorkflowRecord], petUser: UserInfo) = {
+      Future.traverse(externalWorkflowIds) { workflowRec =>
+        // for each workflow query the exec service for status and if has Succeeded query again for outputs
+        toFutureTry(execServiceStatus(workflowRec, petUser) flatMap {
+          case Some(updatedWorkflowRec) => execServiceOutputs(updatedWorkflowRec, petUser)
+          case None => Future.successful(None)
+        })
       }
     }
 
     def queryForWorkflowStatuses() = {
```

Comment:
I can yank out `gimmeTheWorkspaceAndSubmitter` and `gimmeThePetUserInfo` methods, but making a single method that covers most of it would probably require weirdness like "pass in a function that takes a submission id and returns a set of workflows".